app/models/coupon.rb

```diff
@@ -0,0 +1,10 @@
+class Coupon < ApplicationRecord
+
+  belongs_to :user
+
+  validates :code, :start_at, :expire_at, presence: true
+  validates :code, uniqueness: { case_sensitive: false }
+
+  scope :active, -> { where("start_at <= :current_time && expire_at >= :current_time", current_time: DateTime.now) }
```

Comment:
Will the queries be different if I user DateTime.now, DateTime.current, Time.now, Time.current?
Try these out after changing app's timezone to something different than yours